src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -34,9 +34,9 @@ internal static SwitchBinder Create(Binder next, SwitchStatementSyntax switchSyn
                 // In C# 6 and earlier, we use the old binder. In C# 7 and later, we use the new binder which
                 // is capable of binding both the old and new syntax. However, the new binder does not yet
                 // lead to a translation that fully supports edit-and-continue, so it delegates to the C# 6
-                // binder when it can. The "typeswitch" feature flag forces the use of the C# 7 switch binder
+                // binder when it can. The "testV7SwitchBinder" feature flag forces the use of the C# 7 switch binder
                 // for all operations; we use it to enhance test coverage.
-                (parseOptions?.IsFeatureEnabled(MessageID.IDS_FeaturePatternMatching) != false || parseOptions?.Features.ContainsKey("typeswitch") != false)
+                (parseOptions?.IsFeatureEnabled(MessageID.IDS_FeaturePatternMatching) != false || parseOptions?.Features.ContainsKey("testV7SwitchBinder") != false)
```

Comment:
That typically occurs when a user constructs a parse tree and passes it to a speculative semantic model. Since that is already allowed today, it would be a breaking change to disallow it.